### ISSUE TITLE
GMT.jl cannot precompile in macOS

### ISSUE THREAD

**Issue by htyeim** [GenericMappingTools/GMT.jl#627]
The new version of GMT cann't precompile in macOS.

Although I have installed `gmt` using `brew`, the `GMT.jl` still install a new `gmt` using `conda`. The `deps.jl` in `deps` folder contains (the file was generated although I didn't set `INSTALL_GMT`,):
```julia
_GMT_bindir = "/Users/t/.julia/conda/3/bin"
_libgmt  = "/Users/t/.julia/conda/3/lib/libgmt.dylib"
_libgdal = "/Users/t/.julia/conda/3/lib/libgdal.29.dylib"
_libproj = "/Users/t/.julia/conda/3/lib/libproj.22.dylib"
```
The information above is correct (all the files exist).

When I run `using GMT` in julia, however, the error below raise:
```julia
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file: @rpath/libgdal.29.dylib (No such file or directory)
ERROR: LoadError: failed processes:
  Process(`otool -L @rpath/libgdal.29.dylib`, ProcessExited(1)) [1]
  Process(`grep libproj`, ProcessExited(1)) [1]
```

The error correspond the code in `src/GMT.jl`.
```julia
julia> _libgmt = haskey(ENV, "GMT_LIBRARY") ? ENV["GMT_LIBRARY"] : string(chop(read(`gmt --show-library`, String)))
"/Users/t/.julia/conda/3/lib/libgmt.dylib"
julia> _libgdal = string(split(readlines(pipeline(`otool -L $(_libgmt)`, `grep libgdal`))[1])[1])
"@rpath/libgdal.29.dylib"

julia> _libproj = string(split(readlines(pipeline(`otool -L $(_libgdal)`, `grep libproj`))[1])[1])
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file: @rpath/libgdal.29.dylib (No such file or directory)
ERROR: failed processes:
  Process(`otool -L @rpath/libgdal.29.dylib`, ProcessExited(1)) [1]
  Process(`grep libproj`, ProcessExited(1)) [1]
```

And the reason seems related to the lib defined in `libgmt.dylib` (refer to some paths related to `@rpath`) installed by `conda`:
```bash
❯ otool -L /Users/t/.julia/conda/3/lib/libgmt.dylib
/Users/t/.julia/conda/3/lib/libgmt.dylib:
        @rpath/libgmt.6.dylib (compatibility version 6.0.0, current version 6.2.0)
        @rpath/libnetcdf.19.dylib (compatibility version 19.0.0, current version 19.0.0)
        @rpath/libcurl.4.dylib (compatibility version 12.0.0, current version 12.0.0)
        @rpath/libgdal.29.dylib (compatibility version 30.0.0, current version 30.0.0)
        @rpath/libgeos_c.1.dylib (compatibility version 16.0.0, current version 16.2.0)
        @rpath/libpcre.1.dylib (compatibility version 4.0.0, current version 4.12.0)
        @rpath/libfftw3f.3.dylib (compatibility version 10.0.0, current version 10.9.0)
        @rpath/libfftw3f_threads.3.dylib (compatibility version 10.0.0, current version 10.9.0)
        @rpath/liblapack.3.dylib (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate (compatibility version 1.0.0, current version 4.0.0)
        @rpath/libpostscriptlight.6.dylib (compatibility version 6.0.0, current version 6.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
```

The `libgmt.dylib` installed by `brew`, instead, refers to some absolute path:
```
❯ otool -L /usr/local/lib/libgmt.dylib
/usr/local/lib/libgmt.dylib:
	/usr/local/opt/gmt/lib/libgmt.6.dylib (compatibility version 6.0.0, current version 6.2.0)
	/usr/local/opt/netcdf/lib/libnetcdf.19.dylib (compatibility version 19.0.0, current version 19.0.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 9.0.0)
	/usr/local/opt/gdal/lib/libgdal.29.dylib (compatibility version 30.0.0, current version 30.0.0)
	/usr/local/opt/geos/lib/libgeos_c.1.dylib (compatibility version 16.0.0, current version 16.2.0)
	/usr/local/opt/pcre/lib/libpcre.1.dylib (compatibility version 4.0.0, current version 4.12.0)
	/usr/local/opt/fftw/lib/libfftw3f.3.dylib (compatibility version 10.0.0, current version 10.9.0)
	/usr/local/opt/fftw/lib/libfftw3f_threads.3.dylib (compatibility version 10.0.0, current version 10.9.0)
	/usr/local/opt/openblas/lib/libopenblas.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate (compatibility version 1.0.0, current version 4.0.0)
	/usr/local/Cellar/gmt/6.2.0/lib/libpostscriptlight.6.dylib (compatibility version 6.0.0, current version 6.2.0)
```

Is it possible to just use the paths in `deps/deps.jl` (which might be a possible solution)?

**Comment by joa-quim** [comment]
I was afraid of something of this kind. Specially on Macs, but I can't figure out what happened. Tried in a Mac with GMT6.2.0 installed by ``brew`` and all went smoothly.

As you can see in ``GMT.jl get_GMTver()`` the Conda installation is only attempted when a system visible GMT installation is not available. Then it goes to ``catch err1`` branch. And apparently this worked for you. But then again you are reporting errors that occur inside the ``if (!GMTbyConda)`` branch, which should not be visited if GMT was installed by Conda. This is the part that puzzles me more.

To summarize, after upgrading to GMT6.2.0 with brew did you get a warning message telling to run ``GMT.force_precompile()`` and restart Julia?

What happens if you do it again? Then it should be finding the GMT brew installation as it was doing before the GMT upgrade.

**Comment by htyeim** [comment]
> To summarize, after upgrading to GMT6.2.0 with brew did you get a warning message telling to run `GMT.force_precompile()` and restart Julia?

I am not sure... I update all the software using 
`brew update && brew upgrade && brew cu --all --yes --cleanup && mas upgrade && brew cleanup;`, then use ` julia -e "using Pkg; Pkg.update()"`, and never look at the output 🤣 

I think I have solved the problem by removing `gmt` installed by `conda`... I am not sure why `GMT.jl` installed a new `gmt` using `conda` when I have `gmt` installed...
